ingestion/src/metadata/ingestion/processor/pii.py

```diff
@@ -0,0 +1,201 @@
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""
+Processor util to fetch pii sensitive columns
+"""
+import logging
+import re
+import traceback
+from enum import Enum, auto
+
+from presidio_analyzer import AnalyzerEngine
+
+from metadata.generated.schema.entity.data.table import Table, TableData
+from metadata.ingestion.ometa.ometa_api import OpenMetadata
+
+PII = "PII"
+
+
+class PiiTypes(Enum):
+    """PiiTypes enumerates the different types of PII data"""
+
+    NONE = auto()
+    UNSUPPORTED = auto()
+    PHONE = auto()
+    EMAIL = auto()
+    CREDIT_CARD = auto()
+    ADDRESS = auto()
+    ADDRESS_LOCATION = auto()
+    PERSON = auto()
+    LOCATION = auto()
+    BIRTH_DATE = auto()
+    GENDER = auto()
+    NATIONALITY = auto()
+    IP_ADDRESS = auto()
+    SSN = auto()
+    USER_NAME = auto()
+    PASSWORD = auto()
+    ETHNICITY = auto()
+    TAX_ID = auto()
+    KEY = auto()
+    BANKACC = auto()
+    ORGANIZATION = auto()
+    DATE_TIME = auto()
+    PASSPORT = auto()
+    LICENSE = auto()
+
+
+class TagType(Enum):
+    SENSITIVE = "Sensitive"
+    NONSENSITIVE = "NonSensitive"
+
+
+class SpacyEntity(Enum):
+    GPE = PiiTypes.LOCATION.name
+    CREDIT_CARD = PiiTypes.CREDIT_CARD.name
+    EMAIL_ADDRESS = PiiTypes.EMAIL.name
+    IBAN_CODE = PiiTypes.BANKACC.name
+    IP_ADDRESS = PiiTypes.IP_ADDRESS.name
+    NRP = PiiTypes.NATIONALITY.name
+    LOCATION = PiiTypes.LOCATION.name
+    PHONE_NUMBER = PiiTypes.PHONE.name
+    MEDICAL_LICENSE = PiiTypes.LICENSE.name
+    US_DRIVER_LICENSE = PiiTypes.LICENSE.name
+    DATE_TIME = PiiTypes.DATE_TIME.name
+    CARDINAL = PiiTypes.UNSUPPORTED.name
+    URL = PiiTypes.ADDRESS.name
+    US_BANK_NUMBER = PiiTypes.BANKACC.name
+    US_SSN = PiiTypes.SSN.name
+    PERSON = PiiTypes.PERSON.name
+    US_PASSPORT = PiiTypes.PASSPORT.name
+
+
+class NERScanner:
+    """A scanner that uses Spacy NER for entity recognition"""
+
+    def __init__(self):
+        self.text = ""
+        self.analyzer = AnalyzerEngine()
+
+    def get_spacy_to_pii_type(self, spacy_type):
+        return SpacyEntity.__members__.get(spacy_type, PiiTypes.UNSUPPORTED).value
+
+    def get_highest_score_label(self, labels_score):
+        most_used_label_occurrence = 0
+        label_score = None
+        for label, score in labels_score.items():
+            if score[0] == 1.0 and score[1] > len(self.text) * 0.8:
+                return (label, score[0])
+            if score[1] > most_used_label_occurrence:
+                label_score = (label, score[0])
+                most_used_label_occurrence = score[1]
+        return label_score or (None, None)
+
+    def scan(self, text):
+        """Scan the text and return an array of PiiTypes that are found"""
+
+        logging.debug("Processing '%s'", text)
+        pii_tag = f"{PII}.{TagType.NONSENSITIVE.value}"
```

Comment:
I think we can use `fqn.py` to handle this build. Let's try not to hardcode the "." and instead use the FQN module to centralise all the logic there

Reply:
Ok 👍